### PR TITLE
fix: index login state and api request takes too much time

### PR DESCRIPTION
**Pull request by peeyalk**

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -175,31 +175,54 @@ export const getServerSideProps: GetServerSideProps = async (
...
+    channelList.map((channel: string) => {
+      listTimestampInSlackPromises.push(
+        listTimestampInSlack(channel, slackUserToken),
+      );
+    });
+
+    await (
+      await Promise.all(listTimestampInSlackPromises)
+    ).map(({ channel, result }) => {
+      let timestampList: [] = result;
```
Comment (hiroshinishio) [inline pages/index.tsx:208]: let could be "const" in this case??

**Review by hiroshinishio — COMMENTED** [review]
This PR improved getServerSideProps performance from more than 30 seconds to 2.27 seconds
image

File: pages/index.tsx [hunk 2]
```diff
@@ -175,31 +175,54 @@ export const getServerSideProps: GetServerSideProps = async (
...
-          timestampList[y],
-          // @ts-ignore
-          searchQuery,
-          slackUserToken,
+    const numberOfRepliesPromises: any = [];
+    const listTimestampInSlackPromises: any = [];
+
+    channelList.map((channel: string) => {
+      listTimestampInSlackPromises.push(
+        listTimestampInSlack(channel, slackUserToken),
```
[the remark this answers is not in the document]
Comment (peeyalk) [reply pages/index.tsx:201]: here we are pushing the async func in the promises list `listTimestampInSlackPromises` and calling the async promise.all to get the results in an array

```js
await Promise.all(listTimestampInSlackPromises)
```

File: pages/index.tsx [hunk 3]
```diff
@@ -175,31 +175,54 @@ export const getServerSideProps: GetServerSideProps = async (
...
+      );
+    });
+
+    await (
+      await Promise.all(listTimestampInSlackPromises)
```
[the remark this answers is not in the document]
Comment (peeyalk) [reply pages/index.tsx:206]: when wrapping an await for `.map` function you need another await

```js
let result = await some_promise;
result.map(...)
```

```js
await (await some_promise).map(...);
```

these 2 are the same.

File: pages/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (hiroshinishio) [reply pages/index.tsx:201]: Understood.

File: pages/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (peeyalk) [reply pages/index.tsx:206]: But it was unnecessary

[the remark this answers is not in the document]
Comment (hiroshinishio) [reply pages/index.tsx]: Yes, ts says like this

image

**Review by hiroshinishio — APPROVED** [review]

**Pull request merged**